### PR TITLE
CI and rustc warning cleanup

### PR DESCRIPTION
**Pull request by TheBlueMatt**
Fixes a few things about the new Github Actions CI, fixes our CI fuzz stuff, and some rustc compile-time warnings.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 name: Continuous Integration Checks
 
-on:
-  push:
+on: [push, pull_request]
```
Comment (valentinewallace) [inline .github/workflows/build.yml:3]: ah, I had mistakenly assumed `push` would encompass `pull_request`..

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -91,7 +90,7 @@ jobs:
         run: |
           sudo apt-get update
           sudo apt-get -y install build-essential binutils-dev libunwind-dev
-      - name: Fuzz test on Rust ${{ matrix.TOOLCHAIN }}
+      - name: Sanity check fuzz targets on Rust ${{ env.TOOLCHAIN }}
```
Comment (valentinewallace) [inline .github/workflows/build.yml:93]: :+1:

File: fuzz/ci-fuzz.sh [hunk 3]
```diff
@@ -21,6 +21,8 @@ for TARGET in src/bin/*.rs; do
 	HFUZZ_RUN_ARGS="--exit_upon_crash -v -n2"
 	if [ "$FILE" = "chanmon_consistency_target" ]; then
 		HFUZZ_RUN_ARGS="$HFUZZ_RUN_ARGS -F 64 -N100000"
+	elif [ "$FILE" = "full_stack_target" ]; then
```
Comment (valentinewallace) [inline fuzz/ci-fuzz.sh:24]: There's no colors when this script runs in Github Actions :crying_cat_face: (optional, lol)

File: .github/workflows/build.yml (hunk 1 above)
Comment (D4nte) [reply .github/workflows/build.yml:3]: Yep, I just came back to master to fix that as I realised it was needed for fork PRs. Well done @TheBlueMatt :)

**Pull request merged**